packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -682,22 +854,25 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     }
 
     // Position elements
-    double childMainPosition = leadingSpace;
+    double childMainPosition = flipMainAxis ? preferredSize - leadingSpace : leadingSpace;
     child = firstChild;
     while (child != null) {
       final FlexParentData childParentData = child.parentData;
       double childCrossPosition;
       switch (_crossAxisAlignment) {
-        case CrossAxisAlignment.stretch:
         case CrossAxisAlignment.start:
-          childCrossPosition = 0.0;
-          break;
         case CrossAxisAlignment.end:
-          childCrossPosition = crossSize - _getCrossSize(child);
+          childCrossPosition = _startIsTopLeft(flipAxis(direction), textDirection, verticalDirection)
```

Comment:
We shouldn't need to call _startIsTopLeft twice.  We have the value above.  We can just remember it.